test/config-luatest/rpc_test.lua

```diff
@@ -780,3 +803,104 @@ g.test_call_mode = function(g)
+    credentials:
+      users:
+        guest:
+          roles: [super]
+        myuser:
+          password: "secret"
+          roles: [replication]
+          privileges:
+          - permissions: [execute]
+            universe: true
+
+    roles: [one]
+
+    iproto:
+      listen:
+        - uri: 'unix/:./{{ instance_name }}.iproto'
+      advertise:
+        peer:
+          login: 'myuser'
+
+    groups:
+      group-001:
+        replicasets:
+          replicaset-001:
+            instances:
+              instance-001:
+                database:
+                  mode: rw
+              instance-003: {}
+          replicaset-002:
+            instances:
+              instance-002:
+                iproto:
+                  listen:
+                    - uri: 9.0.0.0:8888
+    ]]
+    treegen.write_file(dir, 'config.yaml', config)
+
+    local role = string.dump(function()
+        local function f()
+            return box.info.name
+        end
+
+        rawset(_G, 'f', f)
+
+        return {
+            stop = function() end,
+            apply = function() end,
+            validate = function() end,
+        }
+    end)
+    treegen.write_file(dir, 'one.lua', role)
+
+    local opts = {
+        env = {LUA_PATH = os.environ()['LUA_PATH']},
+        config_file = 'config.yaml',
+        chdir = dir,
+    }
+    g.server_1 = server:new(fun.chain(opts, {alias = 'instance-001'}):tomap())
+    g.server_2 = server:new(fun.chain(opts, {alias = 'instance-002'}):tomap())
+    g.server_3 = server:new(fun.chain(opts, {alias = 'instance-003'}):tomap())
+
+    g.server_1:start({wait_until_ready = false})
+    -- instance-002 intentionally doesn't start.
+    g.server_3:start({wait_until_ready = false})
+
+    g.server_1:wait_until_ready()
+    g.server_3:wait_until_ready()
+
+    local function check_conn()
+        local connpool = require('experimental.connpool')
+        local clock = require('clock')
+
+        -- The connection timeout for filter() and call() methods
+        -- is hardcoded in connpool and equals 10 seconds.
+        local CONNECT_TIMEOUT = 10
+
+        -- call() should wait for any available instance matching
+        -- the requirements and not wait for the unavailable instance
+        -- connection attempts to be timeouted.
+        local opts = { mode = 'alive' }
+        local timestamp_before_call = clock.monotonic()
+        t.assert_equals(connpool.call('f', nil, opts), 'instance-001')
+        t.assert(clock.monotonic() - timestamp_before_call <
+                 CONNECT_TIMEOUT / 2)
+
+        -- call() with some of the prioritized modes (e.g "prefer_ro" or
+        -- "prefer_rw") should try to access all of the instances.
+        opts = { prefer_local = true, mode = 'prefer_ro' }
+        timestamp_before_call = clock.monotonic()
+        t.assert_equals(connpool.call('f', nil, opts), 'instance-003')
+        t.assert(clock.monotonic() - timestamp_before_call >
+                 CONNECT_TIMEOUT / 2)
```

Comment:
Maybe it worth to make this test case a separate one, because an implementation may re-use an information about the recent connection attempt.

If you're worrying about amount of boilerplate code, check out my `cbuilder` and `cluster` helpers. Not an ad :)

Reply:
Thank you. We've discussed the idea of creating a separate test case in #10287. 
`cbuilder` and `cluster` are great, though I avoided them for the consistency with the other tests.

I'm gonna write a separate issue on adding a timeout option for `call()` and `filter()` if you ok with it.

Reply:
Here I meant a separate test case (function), not a separate file. The reason is not an execution time, but a possible re-using of past connect/filter information in a future implementation, so it will be much faster than expected.

Reply:
Oh, I'm sorry. Thanks. Changed it and implemented like this.